mpi-proxy-split/doc/nimrod-build-tutorial.txt

```diff
@@ -0,0 +1,273 @@
+                                        necessary.
+  * conflicting BLAS packages ......... blas-static and openblas-static should
+                                        be necessary. Note that the former
+                                        doesn't exist on CentOS Stream.
+  * PyTables not installed ............ Install PyTables for Python 3:
+                                          python3 -m pip install --user packaging
+                                          python3 -m pip install --user numpy
+                                          python3 -m pip install --user tables
+  * superlu_dist7 built but not found . See the -F flag above more details.
+
+## SUBSEQUENT BUILDS ##
+
+As best written in the DEVELOPMENT WORKFLOW section of nimall/builds:
+
+  Rebuilding with bilder is not required when a change is made to the NIMROD
+  source files. In fact, rebuilding with bilder is also not recommended for
+  development.  If an svn repo is not clean (e.g. nimall/nimdevel has local
+  modifications) bilder will not install the code unless passed the -F flag. If
+  the -F flag is passed the modified code will be installed with the -rXXXXM tag
+  where XXXX is the svn repo revision number.  The softlinks in the $NIMSOFT
+  (install) directory will be updated to point to this installation and thus the
+  binary placed in the PATH variable when nimall.sh is sourced will be updated.
+  
+  The recommended workflow for development is work directly in the <build
+  directory>/nimdevel/<ser/par> directories with debug builds (see below).
+  Execute
+  
+    $ make
+  
+  to rebuild and incorporate your changes. Use
+  
+    $ make -j4
+  
+  to build in parallel with 4 processors. These commands will produce a modified
+  binaries in the <build directory>/nimdevel/<ser/par>/bin directory. One can
+  install these binaries with
+  
+    $ make install
+  
+  however this is not recommended as it will corrupt the installation directory
+  naming system. NIMROD uses a cmake build system and available build targets
+  are shown with
+  
+    $ make help
+  
+  When new source files are added, the corresponding CMakeLists.txt file must be
+  changed in the source code directory and the new file added to the svn repo.
+
+4. Testing NIMROD
+
+### ON CORI ###
+
+Please ask Zhengji Zhao (zzhao@lbl.gov) for access to the test case on Cori.
+
+To run it, navigate to the given directory. Either submit a batch script or run
+it manually.
+
+  $ salloc -N 11 -C haswell -q interactive -t 00:15:00 # customize yourself
+  $ mana_coordinator # optionally; refer to MANA documentation for installation
+  $ srun \
+      --ntasks=88 \
+      --ntasks-per-node=8 \
+      --cpus-per-task=2 \
+      --cpu-bind=cores \
+      MANA_BIN/mana_launch \   # optionally
+      /path/to/nimrod
+
+The test case provided on Cori has to be ran with a multiple of 11 number of
+ranks - this corresponds to the number of tasks.
```

Comment:
-> `The test case provided on Cori has to be ran with the number of MPI ranks being a multiple of 11.  This corresponds to the number of tasks.`